golem-rib/src/compiler/ir.rs

```diff
@@ -60,6 +61,127 @@ pub enum RibIR {
     Negate,
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Encode, Decode)]
+pub enum FunctionReferenceType {
```

Comment:
These look very ugly with no field names.

Also, why do you need this "copy" of the `ParsedFunctionName` type?

Reply:
It's part of the instruction, and they are not copy of ParsedFunctionName.
I can add field names though

Reply:
Added !